geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseQueryPlan.scala

```diff
@@ -55,12 +56,16 @@ case class EmptyPlan(filter: HBaseFilterStrategyType) extends HBaseQueryPlan {
   override def scan(ds: HBaseDataStore): CloseableIterator[SimpleFeature] = CloseableIterator.empty
 }
 
-case class ScanPlan(filter: HBaseFilterStrategyType,
+case class ScanPlan(sft: SimpleFeatureType,
+                    filter: HBaseFilterStrategyType,
                     table: TableName,
                     ranges: Seq[Scan],
                     remoteFilters: Seq[HBaseFilter] = Nil,
                     resultsToFeatures: Iterator[Result] => Iterator[SimpleFeature]) extends HBaseQueryPlan {
   override def scan(ds: HBaseDataStore): CloseableIterator[SimpleFeature] = {
+    val remoteFilters = filter.filter.map { filter =>
+      new JSimpleFeatureFilter(sft, filter)
+    }.toSeq
```

Comment:
Ok, something is at odds here.

There should either be remoteFilters computed here or passed in.  Which way are we going with that?